pkg/injector/patch.go

```diff
@@ -139,41 +135,67 @@ func addContainer(target, add []corev1.Container, basePath string) (patch []JSON
 			path += "/-"
 		}
 		patch = append(patch, JSONPatchOperation{
-			Op:    "add",
+			Op:    addOperation.String(),
 			Path:  path,
 			Value: value,
 		})
 	}
 	return patch
 }
 
-func updateAnnotation(target, add map[string]string, basePath string) (patch []JSONPatchOperation) {
+type operation string
+
+func (o operation) String() string {
+	return string(o)
+}
+
+const (
+	addOperation     operation = "add"
+	replaceOperation operation = "replace"
+)
```

Comment:
I find this unnecessarily complicated, the patch operations are always strings so there doesn't seem to be a good reason why a new type needs to be defined for something as trivial. Please change this to `const strings`